File: app/src/main/java/com/wmods/wppenhacer/xposed/bridge/client/BaseClient.java
package com.wmods.wppenhacer.xposed.bridge.client;

import com.wmods.wppenhacer.xposed.bridge.WaeIIFace;

import java.util.concurrent.CompletableFuture;

public abstract class BaseClient {

    public abstract WaeIIFace getService();

    public abstract CompletableFuture<Boolean> connect();

    public abstract void tryReconnect();

}

File: app/src/main/java/com/wmods/wppenhacer/xposed/bridge/client/BaseClient.kt
package com.wmods.wppenhacer.xposed.bridge.client

import com.wmods.wppenhacer.xposed.bridge.WaeIIFace
import java.util.concurrent.CompletableFuture

abstract class BaseClient {
    abstract val service: WaeIIFace?

    abstract fun connect(): CompletableFuture<Boolean>

    abstract fun tryReconnect()
}
